File: bin/BS.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int key=9;
    int arr[]={6,4,2,1};
    int n=4;
    int l=0,r=n-1;
    bool check=false;
    sort(arr,arr+n);
    while(l<=r)
    {
        int mid=l+(r-l)/2;
        if(arr[mid]==key)
        {
            cout<<"yes";
            check=true;
            break;
        }
        else if(arr[mid]>key)
        {
            r=mid-1;
        }
        else l=mid+1;
    }
    if(!check) {
        cout<<"no";
    }
}

File: quick recap/BS.java
// Online Java Compiler
// Use this editor to write, compile and run your Java code online
import java.util.*;

public class BS {
    public static void main(String[] args) { 
        int[] arr= new int[5];
        // int [] arr={};
        Scanner sc=new Scanner(System.in);
        for(int i=0;i<5;i++)
        {
            arr[i]=sc.nextInt();
        }
        sc.close();
        Arrays.sort(arr);
        int l=0;
        int k=3;
        Boolean check=false;
        int r=arr.length-1;
        while(l<=r)
        {
            int mid=l+(r-l)/2;
            if(k==arr[mid]) {
                System.out.println("found!");
                check=true;
                break;
            }
            else if(k<arr[mid])
            {
                r=mid-1;
            }
            else
            {
                l=mid+1;
            }
        }
        if(!check)
        {
            System.out.println("Not found");
        }       
    }
}